src/Core/Config/Util/ConfigFileManager.php

```diff
@@ -173,7 +173,21 @@ private function loadDataConfig(Cache $config)
 		$filename = $this->configDir . '/' . self::CONFIG_DATA_FILE;
 
 		if (file_exists($filename)) {
-			$dataArray = include $filename;
+
+			$content = '<?php return [];';
+
+			$configStream = fopen($filename, 'r');
+			if (flock($configStream, LOCK_SH)) {
+				$content = fread($configStream, filesize($filename));
+				if (!$content) {
+					throw new ConfigFileException(sprintf('Couldn\'t read file %s', $filename));
+				}
+				flock($configStream, LOCK_UN);
+			}
+
+			fclose($configStream);
+
+			$dataArray = eval('?>' . $content);
```

Comment:
Why the leading `?>` ?

Reply:
Without it, it doesn't work. I read it at stackoverflow tbh *GG*

Reply:
By https://www.php.net/manual/en/function.eval.php :

>  The code must not be wrapped in opening and closing PHP tags, i.e. 'echo "Hi!";' must be passed instead of '<?php echo "Hi!"; ?>'. It is still possible to leave and re-enter PHP mode though using the appropriate PHP tags, e.g. 'echo "In PHP mode!"; ?>In HTML mode!<?php echo "Back in PHP mode!";'.